src/Val.h

```diff
@@ -895,6 +895,11 @@ class TableVal final : public Val, public notifier::detail::Modifiable {
 	// with non-composite index type).
 	ListValPtr ToPureListVal() const;
 
+	// Returns a map of index-to-value's.  The value is nil for sets.
+	// It's up to the caller to Unref() the index Val* when done
+	// with it.
+	std::unordered_map<Val*, ValPtr> ToMap() const;
```

Comment:
Cool!  The only reason I used a bare pointer was indeed because `unordered_map` wasn't happy with an `IntrusivePtr`, and it was beyond my modest skillz to figure out if that could be fixed.

The only problem with you fixing this is that I use a ton of maps in the compiler work, all based on bare pointers, and converting those to use proper `IntrusivePtr`s will be a drudge.  OTOH, it's not that important to do for `const*` pointers, since for them it's clear that there's no memory management issue.

Reply:
Ok, I'll make a separate PR to change this particular `ToMap()` method since figuring out how to make compilers happy with it turned out to be one of those "I guess I actually don't understand C++ that well" moments for me.

Also wasn't thinking to generally suggest changing anything under `script_opt/`, just this one caught my attention due to living in `Val.h`, where people may be more likely to notice and want to use it.